lib/rbac/filterer.rb

```diff
@@ -175,9 +175,6 @@ def search(options = {})
         ids_clause = ["#{klass.table_name}.id IN (?)", target_ids] if klass.respond_to?(:table_name)
       else # targets is a scope, class, or AASM class (VimPerformanceDaily in particular)
         targets = to_class(targets)
-        # could just call all on everything, but that will display deprecation warnings
-        targets = targets.all if targets < ActiveRecord::Base || targets.kind_of?(ActsAsArModel)
-
         scope = apply_scope(targets, scope)
```

Comment:
ooh, `to_class` says if it is a string, convert into a class.
And then call `all` on it to convert into a scope.

So the goal of the code is to get it into what you expect as quickly as possible